File: Data Structures & Algorithms/rotate-array/submission-4.cpp
class Solution {
public:
   
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        if(k>n){
            k = k%n;
        }
    
        unordered_map<int,int> num_index;
        for(int i = 0; i < nums.size();i++){
            num_index[i] = nums[i];
        }
        for(int i = 0, l = k; i < k && l >= 0; i++, l--){
            nums[i] = num_index[n-l];
        }
        int j = k;
        for(int i = 0; i <= n - k && j < n; i++ ){
            nums[j] = num_index[i];
            j++;
        }
    }
};

File: Data Structures & Algorithms/hand-of-straights/submission-4.py
from collections import Counter

class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int):

        if len(hand) % groupSize:
            return False

        nums = Counter(hand)
        avail = set(nums.keys())

        while avail:
            currmin = min(avail)

            i = 0
            while i < groupSize and currmin + i in avail:
                nums[currmin + i] -= 1

                if nums[currmin + i] == 0:
                    avail.remove(currmin + i)

                i += 1

            if i < groupSize:
                return False

        return True